src/go/rpk/pkg/config/weak.go

```diff
@@ -325,7 +325,7 @@ func (c *Config) UnmarshalYAML(n *yaml.Node) error {
 	return nil
 }
 
-func (rpc *RedpandaConfig) UnmarshalYAML(n *yaml.Node) error {
+func (rpc *RedpandaNodeConfig) UnmarshalYAML(n *yaml.Node) error {
 	var internal struct {
 		Directory                  weakString                `yaml:"data_directory"`
```

Comment:
Can you remove AggregateMetrics and DisablePublicMetrics from here too? thanks!

Reply:
Once you do this, the weak_test.go will need a change :D

Reply:
L347-L348 btw

Reply:
Done, thanks